### PR TITLE
fix: fix finally block not executed

### PR DESCRIPTION
**Pull request by wa0x6e**
## Fix

Fix the `finally` not being executed due to error thrown inside catch block. Fixed by returning  a `Promise.reject` instead

## Chores

- Add `status` label to `ipfs_gateways_response_duration_seconds` and `providers_upload_duration_seconds` metrics => differentiate metrics for requests that succeed/failed
- When the error is a well formatted JSON response, send it to Sentry as contexts => avoid error type `Non-Error exception captured with keys: details, reason`

### CODE REVIEW

File: src/providers/index.ts [hunk 1]
```diff
@@ -17,13 +18,14 @@ export default function uploadToProviders(providers: string[], params: any) {
         const size = (params instanceof Buffer ? params : Buffer.from(JSON.stringify(params)))
           .length;
         providersUploadSize.inc({ name, type }, size);
+        status = 1;
 
         return result;
       } catch (e: any) {
         capture(e, { name });
-        throw e;
+        return Promise.reject(e);
```
Comment (Todmy) [inline src/providers/index.ts:26]: it's strange. For this case `throw e;` behaves the same as `return Promise.reject(e);`. And in both cases, the `finally` block will be called

**Review by Todmy — CHANGES_REQUESTED** [review]

File: src/providers/index.ts (hunk 1 above)
Comment (wa0x6e) [reply src/providers/index.ts:26]: Seems like it's not the case, test it with this code 

```
Promise.any(
  [0, 1, 2].map(i => {
    try {
      console.log(i);
      throw new Error('Oops');
    } catch (e) {
      throw e;
      // return Promise.reject(e);
    } finally {
      console.log('Ending ', i);
    }
  })
);
```

Comment (Todmy) [reply src/providers/index.ts:31]: it makes sense. Thank you for explanation

**Review by Todmy — APPROVED** [review]
tACK

**Pull request merged**